src/clims/models/file.py

```
@@ -175,6 +177,34 @@ def _validate_file_path(self):
                                  'with MultiFormatFile(org_file) as wrapped: ...')
 
 
+class ClimsExcelFile:
+    def __init__(self, workbook):
+        self._workbook = workbook
+
+    def __getitem__(self, sheet):
+        return self._workbook[sheet]
+
+    def read_cell(self, cell):
+        """
+        Converts the contents of an excel cell into string or None. Raise an exception
+        for calculated fields.
+        :param cell: openpyxl cell object
+        :return: string or None
+        """
+        # This means that the cell contains a formula, which we do not support.
+        if cell.data_type == 'f':
+            raise ValueError('Excel calculated values are not supported! openpyxl uses cached '
+                             'values for calculated fields, which are populated every last '
+                             'time the doc is opened by MicroSoft Excel software. There is '
+                             'currently no open-source software that evaluates Excel formulas')
+        elif cell.value is None:
+            return None
+        elif isinstance(cell.value, string_types):
+            return cell.value.encode('utf-8')
+        else:
+            return binary_type(cell.value)
```

Does this ever happen? I am hard pressed to think of a case where the data returned would not be possible to encode as unicode. Do you have any example of this? Otherwise perhaps we'd want to raise an exception here instead?

---

I'm not really sure what you mean here

---

regarding cell.value.encode(), I have had problems importing string values, although I don't remember the details unfortunately. encode('utf-8') solved it.

---

regarding cell.binary_type(), as I recall, all int and floats goes in this. when I did "normal" conversion to string I got a message from flake8/pylint (?) to use binary_type instead, which will work in python3 as well.

---

I'm not 100% sure about how oppyxls works here, but if you look at https://six.readthedocs.io/#six.binary_type I think this might cause us some trouble when we switch to Python 3, since I think that the rest of the code expects a string-like type?

---

I've changed it to text_type instead.